auth0/resource_auth0_attack_protection.go

```diff
@@ -259,24 +263,61 @@ func readAttackProtection(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+func updateAttackProtection(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+
+	if ipt := expandSuspiciousIPThrottling(d); ipt != nil {
+		if err := api.AttackProtection.UpdateSuspiciousIPThrottling(ipt); err != nil {
+			return err
+		}
+	}
+
+	if bfp := expandBruteForceProtection(d); bfp != nil {
+		if err := api.AttackProtection.UpdateBruteForceProtection(bfp); err != nil {
+			return err
+		}
+	}
+
+	if bpd := expandBreachedPasswordDetection(d); bpd != nil {
+		if err := api.AttackProtection.UpdateBreachedPasswordDetection(bpd); err != nil {
+			return err
+		}
+	}
+
+	return readAttackProtection(d, m)
+}
+
+func deleteAttackProtection(d *schema.ResourceData, m interface{}) error {
+	d.SetId("")
+	return nil
+}
+
```

Comment:
This is a fairly large deviation from the existing code patterns. I sense a lot of fear and uncertainty within these blocks. Are we sure that we want to necessarily take such defensive action until we have a better sense of what the full scope of the impact is?

Reply:
What makes you sense a lot of fear and uncertainty and what exactly do you mean by existing code patterns? 

I believe we need to be more careful and understand better what this code does. 

On a high level our resource schema is as follows:

```hcl
resource "auth0_attack_protection" "attack_protection" {
    suspicious_ip_throttling {
        // stuff
    }

    brute_force_protection {
        // stuff
    }

    breached_password_detection {
        // stuff
    }
}```

All those 3 attack protection types are optional within the resource. So in case we are omitting `suspicious_ip_throttling` we shouldn't fire a request to update the suspicious ip throttling settings. Before this code change, we were doing exactly that. Unnecessarily. 

The expand function is now looking like:

```go
func expandSuspiciousIPThrottling(d *schema.ResourceData) *management.SuspiciousIPThrottling {
	var ipt management.SuspiciousIPThrottling

	List(d, "suspicious_ip_throttling", IsNewResource(), HasChange()).Elem(func(d ResourceData) {
		// magical things happening inside this block
	})

	return &ipt
}
```

We only set the according `management.SuspiciousIPThrottling` if the resource is actually set within the terraform config file. If it's not set `ipt` will be equal to `nil` and if that's the case we don't even want to fire a request to update the suspicious ip throttling settings. 

So this is what these code changes do. They're an improvement to what we had before, leaving things better than how we found them. 

Let me know if you have any questions around this. Happy to help clarify further to increase confidence.